ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -460,6 +460,20 @@ public String toString() {
     return null;
   }
 
+  @VisibleForTesting
+  public boolean isLeaderReady() {
+    if (!isLeader()) {
```

Comment:
We can remove this check as leaderState == null covers the case and use this function in checkLeaderState line 451 as well.